routes/apiRouter.js

```diff
@@ -19,208 +19,197 @@ routes.get('/', (req, res) => res.json({ message: 'Welcome to Imperial Bin\'s AP
+
+    if(!index) return throwApiError(res, "Please put in an API token!");
     Users.findOne(index, (err, user) => {
-        if (err) return throwApiError(res, "An internal server error occurred! Please contact an admin!")
-        if (user) {
-            const creator = user._id.toString();
-            db.link.loadDatabase()
-            db.link.find({ creator }, (err, documents) => {
-                if (err) return throwApiError(res, "An internal server error occurred! Please contact an admin!");
-                if (documents[0]) {
-                    for (var entry = 0, len = documents.length; entry < len; entry++) {
-                        const _id = documents[entry]._id;
-                        db.link.remove({ _id })
-                        if (documents[entry].imageEmbed && fs.existsSync(`./public/assets/img/${documents[entry].URL}.jpg`)) {
-                            fs.unlinkSync(`./public/assets/img/${documents[entry].URL}.jpg`)
-                        }
-                    }
-                    db.link.loadDatabase();
-                    if (Object.keys(index).indexOf('_id') > -1) return res.redirect('/account')
-                    res.json({
-                        success: true,
-                        message: `Deleted a total of ${documents.length} documents!`,
-                        numberDeleted: documents.length
-                    })
-                } else {
-                    return throwApiError(res, "There was no documents to delete!");
-                }
-            })
-        } else {
-            return throwApiError(res, "Invalid API token!")
-        }
-    })
-})
+        if(err) return throwApiError(res, "Sorry! There was a internal server error, please contact a administrator!");
+        if(!user) return throwApiError(res, "Please put in a valid API token!");
+        const creator = user._id.toString();
+        
+        db.link.loadDatabase();
+        db.findOne({ creator }, (err, documents) => {
+            if(err) return throwApiError(res, "Sorry! There was a internal server error, please contact a administrator!");
+            if(documents.length == 0) return throwApiError(res, "There was no documents to delete!");
+            // Go through every document to delete them.
+            for(const document of documents) {
+                const documentID = document._id;
+                db.link.remove({ documentID });
 
-routes.delete(['/document/:slug', '/deleteCod/:slug', '/paste/:slug'], async (req, res) => {
+                if(document.imageEmbed && fs.existsSync(`./public/assets/img/${document.URL}.jpg`));
+                    fs.unlinkSync(`./public/assets/img/${document.URL}.jpg`);
+            }
+            // (Tech) - I don't see a point in doing this again when
+            // it's already loaded once above? Something i'm missing like
+            // do the new values not update until you load this again?
+            db.link.loadDatabase();
```

Comment:
We need to put db.link.loadDatabase() here because nedb is dumb

Reply:
Understood.